geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/GeoMesaMetadata.scala

```diff
@@ -52,8 +52,8 @@ class AccumuloBackedMetadata(connector: Connector,
                              writeVisibilities: String,
                              authorizationsProvider: AuthorizationsProvider) extends GeoMesaMetadata {
 
+  // warning: only access this map in a synchronized fashion
   private val metaDataCache = new mutable.HashMap[(String, String), Option[String]]()
-    with mutable.SynchronizedMap[(String, String), Option[String]]
```

Comment:
The problem is when we do a filter/map down on line 170. Also, scala's synchronized map impl is being deprecated in favor of just using java's.